src/Aspect/ContractCheckerAspect.php

```diff
@@ -156,7 +164,15 @@ private function isContractSatisfied($instance, $scope, array $args, $annotation
         }
         $instance = is_object($instance) ? $instance : null;
 
-        return $invoker->bindTo($instance, $scope)->__invoke($args, $annotation->value);
+        try {
+            $invocationResult = $invoker->bindTo($instance, $scope)->__invoke($args, $annotation->value);
+        } catch (\Exception $e) {
+            throw new ContractViolation($this->invocation, $annotation->value . " Details: " . $e->getMessage());
+        }
+
+        // if $invocationResult is null, $annotation->value didn't throw any exception
+        // for example - assertion passed (and didn't return bool value)
+        return $invocationResult === null || $invocationResult === true;
```

Comment:
Didn't understand this return part, why don't you return an original `invocationResult`?